crates/swc_ecma_parser/src/parser/util.rs

```diff
@@ -269,7 +269,13 @@ pub(super) trait ExprExt {
                 expr.is_valid_simple_assignment_target(strict)
             }
 
-            Expr::Member(..) | Expr::SuperProp(..) => true,
+            Expr::Member(MemberExpr { ref obj, .. }) => match obj.as_ref() {
+                Expr::Member(..) => obj.is_valid_simple_assignment_target(strict),
```

Comment:
`obj[0]['foo'] = 1` is valid

Reply:
This situation isn't affected, but I still added a test for this.